File: leetcodesln/RemoveElement.cs
namespace leetcodesln
{
    public class RemoveElement
    {
        public int RemoveElementSln(int[] nums, int val)
        {
            // two pointer high low switch

            int l = 0, r = nums.Length - 1;

            while (l <= r)
            {
                if (nums[l] == val)
                {
                    var temp = nums[r];
                    nums[r] = nums[l];
                    nums[l] = temp;
                    --r;
                }
                else
                {
                    l++;
                }
            }
            return r + 1;
        }
    }
}

File: Go/questions/RemoveElement.go
package questions

func RemoveElement(nums []int, val int) int {
	left, right := 0, 0

	for ; right < len(nums); right++ {
		if nums[right] != val {
			nums[left], nums[right] = nums[right], nums[left]
			left++
		}
	}
	return left
}
